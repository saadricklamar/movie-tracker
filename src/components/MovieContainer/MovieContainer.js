import React, { Component } from "react";
import "./MovieContainer.scss";
import { signOut, loadMovies } from "../../actions";
import { key } from "../../util/key";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { fetchMovies } from "../../thunks/fetchMovieThunk";
import MovieCard from "../MovieCard/MovieCard";
import { getFavorites } from "../../util/fetchData";
import { uid } from "react-uid";
import { PropTypes } from 'prop-types';

export class MovieContainer extends Component {
  constructor() {
    super();
    this.state = {
      favorites: false,
      error: "",
      noFavorites: false
    };
  }


componentDidMount = async () => {
  // const favoriteMovies = await getFavorites(this.props.user_id);
  //   this.setState({ favoriteMovies: favoriteMovies.data });
  //   await this.props.loadMovies(url);
  try {
    const response = await fetch(`https://api.themoviedb.org/3/discover/movie?api_key=${key}&sort_by=popularity.desc`);
    if (response.ok) {
      const movies = await response.json();
      this.props.addMovies(movies);
    } else throw Error("Failed to get movies");
  } catch (error) {
    this.setState({ error });
  }
}

  // componentDidMount = async () => {
  //   const url = `https://api.themoviedb.org/3/discover/movie?api_key=${key}&sort_by=popularity.desc`;
  //   
  // };

  toggleFavoritesDisplay = () => {
    const { movies } = this.props
    let isThereAFavorite = movies.filter(movie => movie.favorite === true)
    if (isThereAFavorite.length) {
      this.setState({ favorites: true });
      this.setState({ noFavorites: false });
    } else {
      this.setState({ favorites: false });
      this.setState({ noFavorites: true });
    }
  };

  toggleHomeDisplay = () => {
    this.setState({ favorites: false });
  };

  render() {
    const { movies } = this.props;
    const { favorites } = this.state;
    return (
      <section>
        <header>
          <section className="header-title">
            <h1 className="h1">
              Movie <i className="fas fa-film" /> Tracker
            </h1>
          </section>
        <nav role='navigation'>
          <div id='menuToggle'> 
            <input type="checkbox" />
                <span></span>
                <span></span>
                <span></span>   
            <ul id='menu'>
            <h4 className='user-name'>Welcome, {this.props.user.name}</h4>
              <Link to='/MovieContainer'>
                <li onClick={this.toggleHomeDisplay}>Home</li>
              </Link>
                <hr></hr>
              <Link to='MovieContainer/favorites'>
                <li onClick={this.toggleFavoritesDisplay}>Favorites</li>
              </Link>
                {this.state.noFavorites && (<p className='no-favorites'>You have no favorites</p>)}
                <hr></hr>
              <Link to='/about'>
                <li>About</li>
              </Link>
                <hr></hr>
              <Link to='/Login'>
                <li onClick={signOut}>Sign Out</li>
              </Link>
                <hr></hr>
            </ul>
          </div>
        </nav>
        </header>
        {!favorites && (
          <main className="movies">
            {movies.map(movie => {
              return <MovieCard movie={movie} key={uid(movie)} />;
            })}
          </main>
        )}
        {favorites && (
          <main className="movies">
            {movies.filter(movie => movie.favorite === true).map(movie => {
              return <MovieCard movie={movie} key={uid(movie)} />;
            })}
          </main>
        )}
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  movies: state.movies,
  user_id: state.users.id,
  user: state.users
});

export const mapDispatchToProps = dispatch => ({
  signOut: () => dispatch(signOut()),
  addMovies: obj => dispatch(loadMovies(obj))
});

MovieContainer.propTypes = {
  movies: PropTypes.array.isRequired,
  addMovies: PropTypes.func.isRequired,
  signOut: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  user_id: PropTypes.number.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieContainer);
