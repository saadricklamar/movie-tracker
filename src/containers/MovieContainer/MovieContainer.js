import React, { Component } from "react";
import { signOut, loadMovies, loading } from "../../actions";
import { key } from "../../util/key";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import MovieCard from "../MovieCard/MovieCard";
import { uid } from "react-uid";
import { PropTypes } from "prop-types";

export class MovieContainer extends Component {
  constructor() {
    super();
    this.state = {
      favorites: false,
      error: "",
      noFavorites: false
    };
  }

  componentDidMount = async () => {
    try {
      const response = await fetch(
        `https://api.themoviedb.org/3/discover/movie?api_key=${key}&sort_by=popularity.desc`
      );
      if (response.ok) {
        const movies = await response.json();
        this.props.addMovies(movies);
      } else throw Error("Failed to get movies");
    } catch (error) {
      this.setState({ error });
    }
  };

  toggleFavoritesDisplay = () => {
    const { movies } = this.props;
    let isThereAFavorite = movies.filter(movie => movie.favorite === true);
    if (isThereAFavorite.length) {
      this.setState({ favorites: true });
      this.setState({ noFavorites: false });
    } else {
      this.setState({ favorites: false });
      this.setState({ noFavorites: true });
    }
  };

  toggleHomeDisplay = () => {
    this.setState({ favorites: false });
  };

  render() {
    const { movies, loading } = this.props;
    const { favorites } = this.state;
    if (loading && !movies.length) {
      return (
        <div className="loader">
          <i class="fas fa-ticket-alt"></i>
        </div>
      );
    } else {
      return (
        <section>
          <header>
            <nav role="navigation">
              <div id="menuToggle">
                <input type="checkbox" />
                <span></span>
                <span></span>
                <span></span>
                <ul id="menu">
                  <h4 className="user-name">Welcome, {this.props.user.name}</h4>
                  <Link to="/MovieContainer">
                    <li onClick={this.toggleHomeDisplay}>Home</li>
                  </Link>
                  <hr></hr>
                  <Link to="MovieContainer/favorites">
                    <li onClick={this.toggleFavoritesDisplay}>Favorites</li>
                  </Link>
                  {this.state.noFavorites && (
                    <p className="no-favorites">You have no favorites</p>
                  )}
                  <hr></hr>
                  <Link to="/about">
                    <li>About</li>
                  </Link>
                  <hr></hr>
                  <Link to="/Login">
                    <li onClick={signOut}>Sign Out</li>
                  </Link>
                  <hr></hr>
                </ul>
              </div>
            </nav>
            <section className="header-title">
              <h1 className="h1">
                Movie <i className="fas fa-film" /> Tracker
              </h1>
            </section>
          </header>
          {!favorites && (
            <main className="movies">
              {movies.map(movie => {
                return <MovieCard movie={movie} key={uid(movie)} />;
              })}
            </main>
          )}
          {favorites && (
            <main className="movies">
              {movies
                .filter(movie => movie.favorite === true)
                .map(movie => {
                  return <MovieCard movie={movie} key={uid(movie)} />;
                })}
            </main>
          )}
        </section>
      );
    }
  }
}

export const mapStateToProps = state => ({
  movies: state.movies,
  user_id: state.users.id,
  user: state.users,
  loading: state.loading
});

export const mapDispatchToProps = dispatch => ({
  signOut: () => dispatch(signOut()),
  addMovies: obj => dispatch(loadMovies(obj)),
  loading: bool => dispatch(loading(bool))
});

MovieContainer.propTypes = {
  movies: PropTypes.array.isRequired,
  addMovies: PropTypes.func.isRequired,
  signOut: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  user_id: PropTypes.number.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieContainer);
