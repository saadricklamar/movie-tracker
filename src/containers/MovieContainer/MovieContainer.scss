a {
  text-decoration: none;
  color: white;
  transition: color 0.3s ease;
}

a:hover {
  color: #af2e53;
}

.fa-ticket-alt {
  color: #af2e53;
  padding: 40px;
  font-size: 40px;
}

.no-favorites {
  color: #ff5733;
  font-size: 14px;
  font-weight: normal;
  margin-right: 20px;
  padding-bottom: 5px;
}

.loader {
  margin: 400px auto;
  border: 16px solid #f3f3f3;
  border-top: 16px solid #af2e53;
  border-radius: 50%;
  width: 120px;
  height: 120px;
  animation: spin 2s linear infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

#menuToggle {
  display: block;
  position: relative;
  top: 10px;
  left: 50px;
  z-index: 1;
  -webkit-user-select: none;
  user-select: none;
}

#menuToggle input {
  display: block;
  width: 40px;
  height: 32px;
  position: absolute;
  top: -7px;
  left: -5px;
  cursor: pointer;
  opacity: 0;
  z-index: 2;
  -webkit-touch-callout: none;
}

#menuToggle span {
  display: block;
  width: 33px;
  height: 4px;
  margin-bottom: 5px;
  position: relative;
  background: #cdcdcd;
  border-radius: 3px;
  z-index: 1;
  transform-origin: 4px 0px;
  transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1),
    background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1), opacity 0.55s ease;
}

#menuToggle span:first-child {
  transform-origin: 0% 0%;
}

#menuToggle span:nth-last-child(2) {
  transform-origin: 0% 100%;
}

#menuToggle input:checked ~ span {
  opacity: 1;
  transform: rotate(45deg) translate(-2px, -1px);
  background: white;
}

#menuToggle input:checked ~ span:nth-last-child(3) {
  opacity: 0;
  transform: rotate(0deg) scale(0.2, 0.2);
}

#menuToggle input:checked ~ span:nth-last-child(2) {
  transform: rotate(-45deg) translate(0, -1px);
}

#menu {
  position: absolute;
  height: 350px;
  width: 150px;
  margin: -100px 0 0 -50px;
  padding: 50px;
  padding-top: 125px;
  background: hsla(197, 54%, 41%, 1);
  list-style-type: none;
  -webkit-font-smoothing: antialiased;
  transform-origin: 0% 0%;
  transform: translate(-100%, 0);
  transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1);
}

#menu li {
  padding: 20px 0;
  font-size: 30px;
  margin-right: 25px;
}

#menuToggle input:checked ~ ul {
  transform: none;
}

header {
  padding: 0;
  display: flex;
  justify-content: space-between;
}

.header-title {
  margin: 0 auto;
}

.h1 {
  font-size: 70px;
}

.movies {
  margin-top: 25px;
  display: flex;
  flex-wrap: wrap;
  color: white;
}

main img {
  flex-wrap: wrap;
}

hr {
  color: white;
  width: 100px;
  margin-left: 18px;
}

.user-name {
  color: white;
  font-size: 20px;
  margin-right: 20px;
  font-weight: lighter;
  font-family: "Times-New-Roman";
}

.none {
  color: white;
}

@media (max-width: 700px) {
  .h1 {
    font-size: 40px;
    margin-top: 30px;
  }

  .movies {
    width: 350px;
  }
}

@media (max-width: 500px) {
  .h1 {
    font-size: 35px;
    padding-top: 5px;
  }

  .movies {
    width: 350px;
  }

  #menuToggle {
    left: 10px;
  }
}
