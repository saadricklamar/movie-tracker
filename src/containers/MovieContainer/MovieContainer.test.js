import React from "react";
import ReactDOM from "react-dom";
import { MovieContainer } from "./MovieContainer";
import { shallow } from "enzyme";
import { mapStateToProps, mapDispatchToProps } from "./MovieContainer";

// jest.mock('../../utils/apiCalls')
// jest.mock('../../thunks/fetchMovies')

describe("MovieContainer", () => {
  let wrapper;
  let mockEvent;
  let mockUser = {
    name: "Bridgett"
  };
  let mockUserId = 1
  let mockMovies;
  const mockSignOut = jest.fn();
  const mockLoadMovies = jest.fn();

  beforeEach(() => {
    mockMovies = [
      {
        title: "Avengers",
        vote_average: 10,
        id: 1,
        release_date: 2019,
        overview: "a darn good movie",
        poster_path: undefined
      }
    ];
    mockEvent = { target: { value: "e" } };
    wrapper = shallow(<MovieContainer user={mockUser} movies={mockMovies} user_id={1}/>);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default states", () => {
    const expected = {
      error: "",
      favorites: false,
      favoriteMovies: []
    };

    expect(wrapper.state()).toEqual(expected);
  });

  describe("mapStateToProps", () => {
    it.skip("should call create an object with a movies array", () => {
      const mockState = {
        movies: [mockMovies],
        user_id: {users:1},
        user: {mockUser}
      };
      const expected = {movies: [mockMovies], user_id: {id:1}, user: {mockUser}}

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)

    });
  });
});
