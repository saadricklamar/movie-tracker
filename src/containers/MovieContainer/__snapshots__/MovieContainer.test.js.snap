// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
<section>
  <header>
    <section
      className="header-title"
    >
      <h1
        className="h1"
      >
        Movie 
        <i
          className="fas fa-film"
        />
         Tracker
      </h1>
    </section>
    <nav
      role="navigation"
    >
      <div
        id="menuToggle"
      >
        <input
          type="checkbox"
        />
        <span />
        <span />
        <span />
        <ul
          id="menu"
        >
          <h4
            className="user-name"
          >
            Welcome, 
            Bridgett
          </h4>
          <Link
            to="/MovieContainer"
          >
            <li
              onClick={[Function]}
            >
              Home
            </li>
          </Link>
          <hr />
          <Link
            to="MovieContainer/favorites"
          >
            <li
              onClick={[Function]}
            >
              Favorites
            </li>
          </Link>
          <hr />
          <Link
            to=""
          >
            <li>
              About
            </li>
          </Link>
          <hr />
          <Link
            to="/Login"
          >
            <li
              onClick={[Function]}
            >
              Sign Out
            </li>
          </Link>
          <hr />
        </ul>
      </div>
    </nav>
  </header>
  <main
    className="movies"
  >
    <ConnectFunction
      key="uid1"
      movie={
        Object {
          "id": 1,
          "overview": "a darn good movie",
          "poster_path": undefined,
          "release_date": 2019,
          "title": "Avengers",
          "vote_average": 10,
        }
      }
    />
  </main>
</section>
`;
